addons/sale_project/models/project.py

```diff
@@ -11,6 +11,16 @@
 from odoo.tools.misc import unquote
 
 
+def _confirm_newly_created_order(record, sale_line_id):
+    """ Orders created from project/task are supposed to be confirmed to match the typical flow from sales, but since
+    we allow SO creation from the project/task itself we want to confirm newly created SOs immediately after creation.
+    However this would leads to SOs being confirmed without a single product, so we'd rather do it on record save.
+    """
+    sale_order = record.env['sale.order.line'].browse(sale_line_id).order_id
+    if sale_order and sale_order.state == 'draft':
+        sale_order.action_confirm()
```

Comment:
I would prefer having that function as a method in each model to be able to easily override that in a custom module (if needed)